openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java

```diff
@@ -942,7 +943,9 @@ public void finish(VisitorContext visitorContext) {
                 Map<String, Schema> schemas = openAPI.getComponents().getSchemas();
                 if (CollectionUtils.isNotEmpty(schemas)) {
                     String openApiJson = ConvertUtils.getJsonMapper().writeValueAsString(openAPI);
-                    for (String schemaName : schemas.keySet()) {
+                    // Create a copy of the keySet so that we can modify the map while in a foreach
+                    Set<String> keySet = schemas.keySet().stream().collect(Collectors.toSet());
```

Comment:
don't need to create stream for it. It's not efficient. Just write like this:
```java
Set<String> keySet = new HashSet<>(schemas.keySet());
```

Reply:
@kevind-wgu could you fix it?

Reply:
I assumed that using a stream was efficient as directly using a HashSet. Yes I will change it.

Reply:
@altro3 it is fixed.